lib/src/http/common/util.dart

```diff
@@ -25,7 +32,11 @@ StreamTransformer decodeAttempt(Encoding encoding) {
       subscription = input.listen(
           (data) {
             try {
-              data = encoding.decode(data);
+              if (encode) {
```

Comment:
based on this, if optional params are left at defaults (encode = false + decode = false), this is just a useless passthrough.  Is there any real use case for this fallback?  or can this just take 1 optional `decode` param that defaults to false (encode by default and decode only if specified)?